batch/test/test_batch.py

```diff
@@ -1128,8 +1121,88 @@ def test_update_batch_w_deps_in_update(client: BatchClient):
     status = j2.wait()
 
     assert status['state'] == 'Success', str((status, b.debug_info()))
-    assert j1.job_id == 2, str(b.debug_info())
-    assert j2.job_id == 3, str(b.debug_info())
+    assert b.status()['state'] == 'success', str((b.status(), b.debug_info()))
+    assert b.status()['n_jobs'] == 3, str(b.debug_info())
+
+
+def test_update_batch_w_deps_in_update_always_run(client: BatchClient):
+    bb = client.create_batch()
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    bb.submit()
+
+    j.wait()
+
+    j1 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], always_run=True)
+    j2 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], parents=[j, j1])
+    b = bb.submit()
+    status = j2.wait()
+
+    assert status['state'] == 'Success', str((status, b.debug_info()))
+    assert b.status()['state'] == 'failure', str((b.status(), b.debug_info()))
+
+
+def test_update_batch_w_failing_deps_in_update(client: BatchClient):
+    bb = client.create_batch()
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    bb.submit()
+
+    j1 = bb.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    j2 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], parents=[j, j1])
+    b = bb.submit()
+    j2_status = j2.wait()
+
+    assert j.status()['state'] == 'Success', str((j.status(), b.debug_info()))
+    assert j2_status['state'] == 'Cancelled', str((j2_status, b.debug_info()))
+    assert b.status()['state'] == 'failure', str((b.status(), b.debug_info()))
+
+
+def test_update_with_always_run(client: BatchClient):
```

Comment:
I think the name sells the test a little short here as it's also testing that a job will wait until a parent in another update completes. Not sure if there's a succinct way to express that also in the name here.